### DISCUSSION TITLE
[14.0] No available offline when restarted

### DISCUSSION THREAD

**Discussion by GMagician** [blakeblackshear/frigate#13382] [General]
Not sure if it's a bug or whatever.
When I restart frigate from web page I never get event 'frigate/available=offline' but I always get 'frigate/available=online' when it restart. Is this behaviour intentional?
P.S. I'm using mqtt_monitor to check events

**Comment by hawkeye217** [comment]
The "offline" payload is published by your MQTT broker itself, not Frigate. This is part of MQTT's "last will and testament" when:

- The client disconnects unexpectedly (e.g., due to a network failure or crash)
- The client fails to communicate with the broker within the keep-alive interval

So if Frigate restarts, reconnects, and sends an `online` message before the keep-alive interval defined in your broker is set, you'll never see the `offline` payload.

This is working as intended.

**Reply by GMagician** [reply]
That's not what frigate manuale reports:

> frigate/available
> Designed to be used as an availability topic with Home Assistant. Possible message are: "online": published when Frigate is running (on startup) "offline": published right before Frigate stops

**Reply by hawkeye217** [reply]
The docs need to be updated, then.

**Reply by GMagician** [reply]
Sorry for my stupid question: how can a mqtt broker know how to respond to a message published by a generic client?

**Reply by hawkeye217** [reply]
I'm not sure I understand your question

**Reply by GMagician** [reply]
Well, I try to explain what I don't understand. 'frigate/available' event is published by frigate when it starts adding as payload the string "online". If broker should raise the same message with payload "offiline" how can it know how to publish a message in the exact format?

**Reply by NickM-27** [reply]
because frigate tells it to

https://github.com/blakeblackshear/frigate/blob/f4f3cfa91152952c1a96a9bc721fa3e1b270a9b8/frigate/comms/mqtt.py#L169-L174

**Reply by GMagician** [reply]
ok, then as state in my first post in a not correct/understandable way, my mqqt broker never dispatch message with offline when I restart frigate. Possibly connection is closed too faster than messages is sent?

**Reply by hawkeye217** [reply]
As I mentioned in my first reply, Frigate does not send the message, the broker does. The code above from Frigate instructs the broker, saying "when Frigate goes offline, send this message". The broker has a timeout, so it will not send the offline message until that timeout passes.

See this link for more on the MQTT LWT: https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/

**Reply by GMagician** [reply]
ok now I understood how it works. Thanks

**Reply by bchhabra** [reply]
@hawkeye217  ok so from the above conversation i reduced the keep alive time in my mqtt broker settings to minimum 15secs.
i cannot go below this and seems like restart and reconnnects happens within this 15s time frame,
This make it impossible to broadcast frigate offline message to other mqtt clients.

**Reply by hawkeye217** [reply]
That sounds right. Frigate doesn't take a long time to restart.

**Reply by Unspec7** [reply]
Unsure why, but when I use mosquitto_sub to monitor my broker instance, even after shutting down the container for 10 minutes, frigate/available never publishes an offline message.

**Comment by GMagician** [comment]
@hawkeye217 sorry for bother, can you guide me to source code where /available=online is sent? I did a search on source code but I only found where 'will' is set (lines you already posted)

Thanks

**Reply by hawkeye217** [reply]
https://github.com/blakeblackshear/frigate/blob/f4f3cfa91152952c1a96a9bc721fa3e1b270a9b8/frigate/comms/mqtt.py#L108

**Reply by bchhabra** [reply]
@NickM-27 @hawkeye217 Are you guys planning to do something on this issue?

**Reply by NickM-27** [reply]
what issue are you referring to?

**Reply by bchhabra** [reply]
Right now, this behaviour depends entirely on the message broker's timeout. I suggest defining a minimum wait timeout in the configuration to add a delay before restarting.

**Reply by NickM-27** [reply]
When a user requests a restart they expect Frigate to start restarting immediately, adding a delay would not make sense.

**Reply by bchhabra** [reply]
in this case, messages are not even sent due to timeout, again if you make it configurable, it would be fit for all.
My use-case is I would like to set a variable in my Home assistant when frigate restarts, I have not achieved that so far.

If there are any other possibilities to know when the frigate is restarted I would be happy to know them.

**Reply by NickM-27** [reply]
The timeout is something that you as the user configure on the mqtt broker side. There is nothing that Frigate can do to control this, the broker is responsible for managing and sending this message as the last will.

**Reply by bchhabra** [reply]
On the broker side minimum is 15 secs, at least on Home Assitant MQTT broker does not let you put anything below 15 secs and the frigate restarts within 15 secs.
